pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,93 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+// ValidateSegmentStoreMemorySettings checks whether the user has passed the required values for segment store memory settings.
+// The required values includes: SegmentStoreResources.limits.memory, pravegaservice.cache.size.max , -XX:MaxDirectMemorySize and -Xmx.
+// Once the required values are set, the method also checks whether the following conditions are met:
+// Total Memory > JVM Heap (-Xmx) + JVM Direct Memory (-XX:MaxDirectMemorySize)
+// JVM Direct memory > Segment Store read cache size (pravegaservice.cache.size.max).
+func (p *PravegaCluster) ValidateSegmentStoreMemorySettings() error {
+	if p.Spec.Pravega.SegmentStoreResources == nil {
+		return fmt.Errorf("spec.pravega.segmentStoreResources cannot be empty")
+	}
+
+	if p.Spec.Pravega.SegmentStoreResources.Limits == nil {
+		return fmt.Errorf("spec.pravega.segmentStoreResources.limits cannot be empty")
+	}
+
+	if p.Spec.Pravega.SegmentStoreResources.Requests == nil {
+		return fmt.Errorf("spec.pravega.segmentStoreResources.requests cannot be empty")
+	}
```

Comment:
The PR changes in `pravega.go` do allow for the requests not being set.
The better way to deal with the situation is to set `totalMemoryRequests` to zero if `segmentStoreResources.requests` is not set (otherwise proceed normally).